### PR TITLE
Add  json-rpc-relay metrics

### LINKED ISSUE #324: Add Metrics for json-rpc-relay
### Problem

Currently there are no metrics for the json-rpc-relay service

### Solution

- Add target to prometheus configuration
- Add Dashboard to Grafana default dashboards

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by AlfredoG87**
**Description**:
1. Adding the `json-rpc-relay` as "target" for metrics scrapping by prometheus
2. Adding the current dashboard with some modifications for Hedera Local Node for displaying the `json-rpc-relay` metrics

**Related issue(s)**:

Fixes #324 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: compose-network/prometheus/prometheus.yml [hunk 1]
```diff
@@ -2,6 +2,13 @@ global:
   scrape_interval: 3s
 
 scrape_configs:
+  - job_name: 'relay-development'
+    scrape_interval: 30s
+    static_configs:
+    - targets: [ 'json-rpc-relay:7546' ]
```
Comment (Nana-EC) [inline compose-network/prometheus/prometheus.yml:8]: Q: Should this capture the `/metrics` path?

File: compose-network/grafana/dashboards/json-rpc-relay/default-prom.json [hunk 2]
```diff
@@ -0,0 +1,4944 @@
+{
+    "annotations": {
```
Comment (Nana-EC) [inline compose-network/grafana/dashboards/json-rpc-relay/default-prom.json:2]: nit: File isn't properly indented.

**Review by Nana-EC — COMMENTED** [review]
Suggestion and question

**Comment by AlfredoG87** [comment]
> Nice addition. I just have a few questions:
> 
> * Is the static ip assignment for the relay necessary?  Something like this should work:
> 
> ```yaml
>     static_configs:
>       - targets: ['json-rpc-relay:7546']
> ```
> 
> * Maybe the dashboard should be named something else, or moved to a different subdirectory. Grafana has a limitation where the name of the dashboard can't match the directory it's placed in.

Yes, I also think is not necessary, but @ebadiere encountered the issue on his local that was not able to reach it using the name as suggested `json-rpc-relay` and am trying to figure out why is it failing the CI tests so thought it might be a good chance to try that out, but yeah, on a previous commit, before I had it only with the name, and locally it worked. but fails on the CI. 

Let me take a look at your other suggestion and make the appropriate change to the folder 👍  thanks!!
Also, might return it to use the name as suggested to simplify the amount of proposed changes, anyways it did not resolve my issue with the CI so better to rollback that and not assign a static IP to the relay.

**Comment by isavov** [comment]
@AlfredoG87 I've discovered an intermittent issue with local node that sometimes affects runs in the CI. It's been fixed via this PR #344. 
It seems you are affected by the same issues. I would suggest rebasing on main to see if the issue is resolved for you.

**Comment by AlfredoG87** [comment]
Thank you @isavov, will try it later today 👍

File: compose-network/prometheus/prometheus.yml (hunk 1 above)
Comment (AlfredoG87) [reply compose-network/prometheus/prometheus.yml]: Yes, it captures that path by default, no need to specify the `/metrics` path explicitly

**Comment by AlfredoG87** [comment]
@isavov Thank you, indeed that rebase fixed my issue, also simplified the changes to a minimum. 👍

**Review by Nana-EC — APPROVED** [review]
LGTM

**Review by isavov — APPROVED** [review]
Nice addition to the local node!
LGTM

File: compose-network/grafana/dashboards/json-rpc-relay/default-prom.json [hunk 3]
```diff
@@ -0,0 +1,6044 @@
...
+        "targets": [
+          {
+            "datasource": {
+              "type": "prometheus",
+              "uid": "local-prom"
+            },
+            "editorMode": "code",
+            "expr": "avg (rate(rpc_relay_method_response_sum{cluster=\"$cluster\", namespace=\"$namespace\"}[$__rate_interval])) / avg (rate(rpc_relay_method_response_count{cluster=\"$cluster\", namespace=\"$namespace\"}[$__rate_interval]))",
+            "hide": false,
+            "legendFormat": "HashIO E2E Avg. Latency",
```
Comment (isavov) [inline compose-network/grafana/dashboards/json-rpc-relay/default-prom.json:3883]: nit: Maybe this should be renamed to something like "Relay E2E Avg. Latency"

File: compose-network/grafana/dashboards/json-rpc-relay/default-prom.json [hunk 4]
```diff
@@ -0,0 +1,6044 @@
...
+            "datasource": {
+              "type": "prometheus",
+              "uid": "local-prom"
+            },
+            "editorMode": "code",
+            "exemplar": false,
+            "expr": "avg (rate(rpc_relay_method_response_sum{cluster=\"$cluster\", namespace=\"$namespace\"}[$__rate_interval])) / avg (rate(rpc_relay_method_response_count{cluster=\"$cluster\", namespace=\"$namespace\"}[$__rate_interval]))",
+            "hide": false,
+            "instant": true,
+            "legendFormat": "HashIO",
```
Comment (isavov) [inline compose-network/grafana/dashboards/json-rpc-relay/default-prom.json:3769]: nit: Maybe rename to Relay

File: compose-network/grafana/dashboards/json-rpc-relay/default-prom.json (hunk 3 above)
Comment (AlfredoG87) [reply compose-network/grafana/dashboards/json-rpc-relay/default-prom.json:3883]: Good suggestions!

**Pull request merged**